### ISSUE TITLE
Spotify playlist not remembered

### ISSUE THREAD

**Issue by nagyrobi** [music-assistant/hass-music-assistant#153]
Steps to reproduce:
- Choose (through media browser button/popup on the player) a spotify playlist to play
- Close the popup window, and open media browser again
- Choose from _another playlist_ just a track to play directly (not playlist play on top, just track directly (use case: you don't want to start with the first track))
- Close the popup window
- Press Next on the player: it will jump to the next track of the playlist selected in the first step.

Expected behavior would be to remain on the playlist from which the track way played in the second step.

**Comment by marcelveldt** [comment]
I'm not sure I understand it. Can you support the above with some screenshots perhaps ?
With media browser you mean the MA interface or the HA media browser ?
The Spotify playlist is not remembered, it is added to the queue. You can even add multiple playlists to the queue (heck, even from different providers) and shuffle tracks if you want.

**Comment by nagyrobi** [comment]
Choose (through media browser button/popup on the player) a spotify playlist to play:
![kép](https://user-images.githubusercontent.com/1550668/168228894-3c21649d-ad90-4d1f-88fe-29c17b38a7d2.png)
![kép](https://user-images.githubusercontent.com/1550668/168229182-db7249af-6172-4fc7-9773-d6b2d4b61694.png)
![kép](https://user-images.githubusercontent.com/1550668/168229298-ea9a4cbb-2faa-4a2f-a0a7-568db9c8e02c.png)
![kép](https://user-images.githubusercontent.com/1550668/168229591-921add79-fbeb-4852-a9cb-a923adcc004c.png)

Close the popup window, and open media browser again:
![kép](https://user-images.githubusercontent.com/1550668/168230122-f5e5860d-643f-48ba-9770-1cc0920c9ea8.png)
![kép](https://user-images.githubusercontent.com/1550668/168230397-3aafb94a-11fd-49d0-95ce-b1c144acc1d9.png)
![kép](https://user-images.githubusercontent.com/1550668/168229182-db7249af-6172-4fc7-9773-d6b2d4b61694.png)
![kép](https://user-images.githubusercontent.com/1550668/168229298-ea9a4cbb-2faa-4a2f-a0a7-568db9c8e02c.png)

Choose from another playlist just a track to play directly (not playlist play on top, just track directly (use case: you don't want to start with the first track)):
![kép](https://user-images.githubusercontent.com/1550668/168230960-69481671-cdaf-4c27-84d8-737d58219999.png)
![kép](https://user-images.githubusercontent.com/1550668/168231143-d8ef9637-cce4-457f-bc61-1aecdf87db40.png)
(**note**: I chose the 3rd track)

Close the popup window:
![kép](https://user-images.githubusercontent.com/1550668/168231400-8071569a-31f4-44c4-b6fc-351edcc22899.png)

Press Next on the player: it will jump to the next track of the playlist selected in the first step.:
![kép](https://user-images.githubusercontent.com/1550668/168231672-8078f30e-dfa4-46af-ba3b-b8d6941d60a6.png)
![kép](https://user-images.githubusercontent.com/1550668/168231849-8f073c07-b91a-436c-a5a9-2a30f8c8eb26.png)

Track played next belongs to the previous playlist (but with the **index=4** from the second one):
![kép](https://user-images.githubusercontent.com/1550668/168232151-7b954530-9f33-46ef-9326-12e654732042.png)

**Comment by nagyrobi** [comment]
> The Spotify playlist is not remembered, it is added to the queue.

That's the issue it seems. In the scenario depicted above, it should clear the queue first, as the user expects to stay on the last playlist chosen, everything that was loaded before should be discarded.

**Comment by marcelveldt** [comment]
Now I understand. Well that's a matter of preference. In most music players/systems, if you select a single track it will add THAT TRACK to the queue. If you want to play the playlist, you play the playlist (the button on top).

Problem is that the Music Assistant panel has more fine grained control. The Home Assistant default media browser is very basic and does not bring this kind of choices. If this is requested more often I could make a setting for it with the default behavior it should follow.

So, in short this is not a bug but by design and a very common design choice in music software.
I will treat this as a feature request to be able to set the default "play" behavior for the builtin media browser.

**Comment by nagyrobi** [comment]
> if you select a single track it will add THAT TRACK to the queue

Well I did not just selected it (for later playback) but I played it immediately. It's pretty obvious from the user perspective that by that I also chose a new playlist but I didn't want to start with the first item from it but with some other one.

**Comment by marcelveldt** [comment]
I understand what you mean but that is a personal preference. It is not implemented that way currently but we can create some sort of config toggle to set this behavior. Current default when you click on a track is "play" which basically means: insert this track on the top of the queue. What you want is a question if you want to play the track of the playlist or add the track somewhere on the queue.

**Comment by marcelveldt** [comment]
Created a feature request for this one.
https://github.com/music-assistant/hass-music-assistant/discussions/513
